sdk/storage/storage-queue/review/generated/models/index.ts

```diff
@@ -0,0 +1,1450 @@
+  allowedHeaders: string;
+  /**
+   * The response headers that may be sent in the response to the CORS request and exposed by the
+   * browser to the request issuer
+   */
+  exposedHeaders: string;
+  /**
+   * The maximum amount time that a browser should cache the preflight OPTIONS request.
+   */
+  maxAgeInSeconds: number;
+}
+
+/**
+ * An interface representing GeoReplication.
+ */
+export interface GeoReplication {
+  /**
+   * The status of the secondary location. Possible values include: 'live', 'bootstrap',
+   * 'unavailable'
+   */
+  status: GeoReplicationStatusType;
+  /**
+   * A GMT date/time value, to the second. All primary writes preceding this value are guaranteed
+   * to be available for read operations at the secondary. Primary writes after this point in time
+   * may or may not be available for reads.
+   */
+  lastSyncTime: Date;
+}
+
+/**
+ * the retention policy
+ */
+export interface RetentionPolicy {
+  /**
+   * Indicates whether a retention policy is enabled for the storage service
+   */
+  enabled: boolean;
+  /**
+   * Indicates the number of days that metrics or logging or soft-deleted data should be retained.
+   * All data older than this value will be deleted
+   */
+  days?: number;
+}
+
+/**
+ * Azure Analytics Logging settings.
+ */
+export interface Logging {
+  /**
+   * The version of Storage Analytics to configure.
+   */
+  version: string;
+  /**
+   * Indicates whether all delete requests should be logged.
+   */
+  deleteProperty: boolean;
+  /**
+   * Indicates whether all read requests should be logged.
+   */
+  read: boolean;
+  /**
+   * Indicates whether all write requests should be logged.
+   */
+  write: boolean;
+  retentionPolicy: RetentionPolicy;
+}
+
+/**
+ * An interface representing StorageError.
+ */
+export interface StorageError {
+  message?: string;
+}
+
+/**
+ * An interface representing Metrics.
+ */
+export interface Metrics {
+  /**
+   * The version of Storage Analytics to configure.
+   */
+  version?: string;
+  /**
+   * Indicates whether metrics are enabled for the Queue service.
+   */
+  enabled: boolean;
+  /**
+   * Indicates whether metrics should generate summary statistics for called API operations.
+   */
+  includeAPIs?: boolean;
+  retentionPolicy?: RetentionPolicy;
+}
+
+/**
+ * A Message object which can be stored in a Queue
+ */
+export interface QueueMessage {
+  /**
+   * The content of the message
+   */
+  messageText: string;
+}
+
+/**
+ * The object returned in the QueueMessageList array when calling Get Messages on a Queue.
+ */
+export interface DequeuedMessageItem {
+  /**
+   * The Id of the Message.
+   */
+  messageId: string;
+  /**
+   * The time the Message was inserted into the Queue.
+   */
+  insertionTime: Date;
+  /**
+   * The time that the Message will expire and be automatically deleted.
+   */
+  expirationTime: Date;
+  /**
+   * This value is required to delete the Message. If deletion fails using this popreceipt then the
```

Comment:
May be useful to add a short description of what popReceipt is. In the current comment it is clear that this is required but I'm having a hard time understanding what this is.

Reply:
It's storage related concepts. Could add more comments into swagger for generated models.

---
In reply to: [334185676](https://github.com/Azure/azure-sdk-for-js/pull/5476#discussion_r334185676) [](ancestors = 334185676)

Reply:
Fix in swagger.

Reply:
Logged #5562